src/inspector_socket.cc

```diff
@@ -542,11 +559,10 @@ static int message_complete_cb(http_parser* parser) {
     int len = sizeof(accept_response);
     if (write_to_client(inspector, accept_response, len) >= 0) {
       handshake_complete(inspector);
+      inspector->http_parsing_state = nullptr;
```

Comment:
It didn't, but I still refactored the code to introduce a boolean flag. Hope this makes the code more maintainable.